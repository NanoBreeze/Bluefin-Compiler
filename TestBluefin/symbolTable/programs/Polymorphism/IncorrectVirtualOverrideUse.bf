/*
Verifies that the 'override' and 'virtual' keywords can only be applied to methods. 
We also can't apply 'override' to a method in a class without a superclass
*/


struct FakeDer {
	void f(int a, bool b) override{} // no superclass, cannot use override
};

float g() override {}				// 'override' keyword may only appear in methods
virtual float h() {}				// 'virtual' keyword may only appear in methods