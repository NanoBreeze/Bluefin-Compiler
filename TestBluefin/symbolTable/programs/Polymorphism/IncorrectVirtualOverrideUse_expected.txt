Cannot apply 'override' qualifier on method 'f' since its class has no parent class
'override' specifier can be applied only to methods. 'g' is a function
'virtual' specifier can be applied only to methods. 'h' is a function
